lib/smart_proxy_pulp_plugin/disk_usage.rb

```diff
@@ -0,0 +1,52 @@
+module PulpProxy
+  class DiskUsage
+    include ::Proxy::Util
+    SIZE = { :kilobyte => 1_024, :megabyte => 1_048_576, :gigabyte => 1_073_741_824, :terabyte => 1_099_511_627_776 }
+
+    attr_reader :path, :stat, :size
+
+    def initialize(opts ={})
+      raise(::Proxy::Error::ConfigurationError, 'Unable to continue - must provide a path.') if opts[:path].nil?
+      @pathes_hash = path_hash(opts[:path])
+      @path = @pathes_hash.values.join(' ')
+      @size = SIZE[opts[:size]] || SIZE[:gigabyte]
+      @stat = {}
+      find_df
+      get_stat
+    end
+
+    def to_json
+      stat.to_json
+    end
+
+    private
+
+    attr_reader :command_path
+
+    def find_df
+      @command_path = which('df') || raise(::Proxy::Error::ConfigurationError, 'df command was not found unable to retrieve usage information.')
+    end
+
+    def command
+      "LANG=C #{command_path} -B#{size} #{path}"
```

Comment:
Additional tip, if you struggle getting it working, make sure the STDERR is visible. E.g. if you accidentaly send something incorrectly, nothing is printed on STDOUT (but STDERR).

```ruby
Open3.popen3({"LC_ALL" => "C"}, "/usr/bin/df", "-B KB", "/home") {|i,o,e,t|
puts e.read
}
>>> /usr/bin/df: invalid -B argument ' KB'
```

Apparently, good practice is to redirect STDERR into logger.error, please do that.

Here is a working example for your scenario in a variable with kwargs syntax (not sure how this is called in Ruby :-)

```ruby
?> Open3.popen3({"LC_ALL" => "C"}, *opts) {|i,o,e,t|
?>   puts o.read
>> }
Filesystem                                             1kB-blocks       Used   Available Use% Mounted on
/dev/mapper/luks-5cb695a3-6657-44f4-8297-a571e02e5492 186676081kB 23477838kB 153691960kB  14% /home
```